0-SCore/Scripts/UtilityAI/UAITaskApproachSpotSDX.cs

```diff
@@ -0,0 +1,121 @@
+﻿namespace UAI
+{
+    /// <summary>
+    /// Utility AI task to approach a spot, as determined by the investigate position.
+    /// The name and functionality is based on <see cref="EAIApproachSpot"/>.
+    /// </summary>
+    public class UAITaskApproachSpotSDX : UAITaskBase

```

Comment:
I know you also have something similar called `UAITaskMoveToInvestigate` but it doesn't really handle the same cases, plus it doesn't always choose to go to the investigate position, which is necessary here.